src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -32,13 +32,34 @@ internal ModelOperationsCatalog(IHostEnvironment env)
         /// <param name="stream">A writeable, seekable stream to save to.</param>
         public void Save(ITransformer model, Stream stream) => model.SaveTo(_env, stream);
 
+        public void Save<TSource>(IDataLoader<TSource> model, Stream stream)
+        {
+            using (var rep = RepositoryWriter.CreateNew(stream))
+            {
+                ModelSaveContext.SaveModel(rep, model, "Model");
+                rep.Commit();
+            }
+        }
+
+        public void Save<TSource>(IDataLoader<TSource> loader, ITransformer model, Stream stream) =>
+            Save(new CompositeDataLoader<TSource, ITransformer>(loader, new TransformerChain<ITransformer>(model)), stream);
+
         /// <summary>
         /// Load the model from the stream.
         /// </summary>
         /// <param name="stream">A readable, seekable stream to load from.</param>
         /// <returns>The loaded model.</returns>
         public ITransformer Load(Stream stream) => TransformerChain.LoadFrom(_env, stream);
 
+        public IDataLoader<IMultiStreamSource> LoadAsCompositeDataLoader(Stream stream)
```

Comment:
>public IDataLoader<IMultiStreamSource> LoadAsCompositeDataLoader(Stream stream) [](start = 8, length = 79)

Should this return `CompositeDataLoader` instead of `IDataLoader`?  The name is: `LoadAsCompositeDataLoader`, which makes me think it should return a `CompositeDataLoader`